libs/category/state/src/reducers/category/category.reducer.spec.ts

```diff
@@ -36,40 +64,102 @@ import { daffCategoryReducer } from './category.reducer';
 describe('Category | Category Reducer', () => {
 
   let categoryFactory: DaffCategoryFactory;
-  let categoryPageConfigurationStateFactory: DaffStatefulCategoryPageConfigurationStateFactory;
+  let categoryPageMetadataFactory: DaffCategoryPageMetadataFactory;
+  let equalFilterFactory: DaffCategoryFilterEqualFactory;
+  let equalOptionFactory: DaffCategoryFilterEqualOptionFactory;
+  let rangeFilterFactory: DaffCategoryFilterRangeNumericFactory;
+  let rangePairFactory: DaffCategoryFilterRangeNumericPairFactory;
+  let equalFilterRequestFactory: DaffCategoryFilterRequestEqualFactory;
+  let equalFilterToggleRequestFactory: DaffCategoryFilterToggleRequestEqualFactory;
+  let rangeFilterRequestFactory: DaffCategoryFilterRequestRangeNumericFactory;
+  let rangeFilterToggleRequestFactory: DaffCategoryFilterToggleRequestRangeNumericFactory;
+  let rangeFilterRequestOptionFactory: DaffCategoryFilterRangeNumericRequestOptionFactory;
+
+  let initialState: DaffCategoryReducerState;
+  let categoryPageMetadata: DaffCategoryPageMetadata;
   let category: DaffCategory;
-  let categoryPageConfigurationState: DaffStatefulCategoryPageConfigurationState;
+  let currentEqualFilter: DaffCategoryEqualFilter;
+  let currentAppliedEqualFilterOption: DaffCategoryFilterEqualOption;
+  let currentUnappliedEqualFilterOption: DaffCategoryFilterEqualOption;
+  let currentRangeFilter: DaffCategoryFilterRangeNumeric;
+  let currentRangeFilterPair: DaffCategoryFilterRangePair<number>;
+  let currentRangeFilterPairLabel: string;
+  let equalFilterRequest: DaffCategoryFilterEqualRequest;
+  let equalFilterToggleRequest: DaffToggleCategoryFilterEqualRequest;
+  let rangeFilterRequest: DaffCategoryFilterRangeRequest;
+  let rangeFilterToggleRequest: DaffToggleCategoryFilterRangeNumericRequest;
+  let rangeFilterRequestOption: DaffCategoryFilterRangeRequestOption<number>;
+  let rangeFilterRequestOptionLabel: string;
   let categoryId: string;
-  const initialState: DaffCategoryReducerState = {
-    categoryPageConfigurationState: {
-      id: null,
-      filter_requests: [],
-      applied_sort_option: null,
-      applied_sort_direction: null,
-      current_page: null,
-      page_size: null,
-      total_pages: null,
-      filters: [],
-      sort_options: {
-        default: null,
-        options: [],
-      },
-      total_products: null,
-      product_ids: [],
-      daffState: DaffState.Stable,
-    },
-    categoryLoading: false,
-    productsLoading: false,
-    errors: [],
-  };
 
   beforeEach(() => {
-    categoryFactory = new DaffCategoryFactory();
-    categoryPageConfigurationStateFactory = new DaffStatefulCategoryPageConfigurationStateFactory();
+    categoryFactory = TestBed.inject(DaffCategoryFactory);
```

Comment:
This before each is rather involved. Can you instead set initial state specifically as required by each reducer and then only factory out what precisely needs to be tested?

Reply:
moved